### PR TITLE
fix: gas settings dialog

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

### CODE REVIEW

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/TradeForm.tsx [hunk 1]
```diff
@@ -443,7 +443,11 @@ export const TradeForm = memo<AllTradeFormProps>(
                                     className={classes.icon}
                                     size="small"
                                     onClick={async () => {
-                                        const { slippageTolerance, transaction } = await selectAdvancedSettings()
+                                        const { slippageTolerance, transaction } = await selectAdvancedSettings({
+                                            transaction: {
+                                                gas: focusedTrade?.gas.value ?? 150000,
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/TradeForm.tsx]: CC @nuanyang233

File: packages/shared/src/UI/components/SettingsBoard/GasForm.tsx [hunk 2]
```diff
@@ -78,34 +100,43 @@ export function GasForm(props: GasFormProps) {
         'maxFeePerGas',
         'maxPriorityFeePerGas',
     ])
+    const [gasPriceByUser, setGasPriceByUser] = useState<string>()
+    const [maxFeePerGasByUser, setMaxFeePerGasByUser] = useState<string>()
+    const [maxPriorityFeePerGasByUser, setMaxPriorityFeePerGasByUser] = useState<string>()
 
-    const error = useMemo(() => {
+    const errorCenter = useMemo(() => {
```
Comment (UncleBill) [inline packages/shared/src/UI/components/SettingsBoard/GasForm.tsx:109]: Access `methods.formState.errors` many times.
```suggestion
    const errors = methods.formState.errors
    const errorCenter = useMemo(() => {
```

File: packages/shared/src/UI/components/SettingsBoard/GasSection.tsx [hunk 3]
```diff
@@ -72,27 +93,11 @@ export function GasSection(props: GasSectionProps) {
...
-                                                  | undefined) ?? 0,
-                                          ).toFixed(2)
-                                        : 0}
-                                    {' Gwei'}
-                                </span>
-                            )}
+                            <span className={classes.price}>
+                                {' '}
+                                {customPrice}
+                                {' Gwei'}
```
Comment (UncleBill) [inline packages/shared/src/UI/components/SettingsBoard/GasSection.tsx:99]:
```suggestion
                                {` ${customPrice} Gwei`}
```

File: packages/shared/src/UI/components/SettingsBoard/GasSection.tsx [hunk 4]
```diff
@@ -56,10 +57,30 @@ export function GasSection(props: GasSectionProps) {
...
+            activeTab === GasSettingsType.Basic
+                ? suggestedMaxFeePerGas ?? 0
+                : formatWeiToGwei(
+                      isEIP1559 ? maxFeePerGas ?? suggestedMaxFeePerGas ?? 0 : gasPrice ?? suggestedMaxFeePerGas ?? 0,
+                  ),
+            2,
+        ),
+        2,
+        2,
+    )
```
Comment (UncleBill) [inline packages/shared/src/UI/components/SettingsBoard/GasSection.tsx:82]: A bit messy.
Can we avoid type assertion here?.

File: packages/shared/src/UI/components/SettingsBoard/GasSection.tsx (hunk 3 above)
Comment (guanbinrui) [reply packages/shared/src/UI/components/SettingsBoard/GasSection.tsx:99]: Hmmm, we disallowed template strings in JSX.

**Pull request merged**